content/en/blog/performance-benchmarks-gke.md

```diff
@@ -0,0 +1,354 @@
+---
+title: gRPC performance benchmarks on GKE
+date: 2022-02-17
+authors:
+  - name: Paulo Castello da Costa
+    link: https://github.com/paulosjca
+  - name: Jan Tattermusch
+    link: https://github.com/jtattermusch
+---
+
+[gRPC performance benchmarks] have now been transitioned to run on GKE, with
+similar results but much increased flexibility.
+
+## Background
+
+gRPC performance testing requires a test driver and workers (one or more clients
+and a server), as described in [gRPC performance benchmarks]. Each test may have
+a different configuration, or _scenario_, that is passed to the driver and
+specified as a JSON file. Previously, the driver was run by the continuous
+integration process, and the workers were run on long-lived GCE VMs. This gave
+rise to several limitations:
+
+1. Tests ran sequentially and were difficult to parallelize, since they ran on
+   (the same) fixed VMs.
+
+1. The state of the VMs was not guaranteed to be the same at the start of each
+   test.
+
+1. Running manual experiments required configuring new VMs, which was a manual
+   process, or reusing existing VMs, with the risks of collision with other
+   users and of having VMs in an unknown state.
+
+## Benchmarking on Kubernetes
+
+The core of the current framework is a [custom controller] for managing
+Kubernetes resources of kind [LoadTest]. This controller must be deployed to a
+Kubernetes cluster before load tests can be run on it. The controller is
+implemented with [kubebuilder]. The code for the controller is stored in the
+[Test Infra] repository. For further documentation on individual LoadTest
+fields, see the [LoadTest implementation].
+
+LoadTest configurations specify driver, client and server pods to be created for
+the test. Once the configuration is applied to the cluster (for instance, with
+`kubectl apply -f`), the controller will create the pods and the test will run.
+If multiple configurations are applied to the cluster, the controller will
+create pods as long as there are resources available, allowing tests to run in
+parallel.
+
+[Examples] include basic configurations that can be applied directly, and
+templates that require additional steps and parameter substitution.
+
+- The basic configurations for each supported language rely on a default worker
+  image that is bundled with each release of the controller, and build the gRPC
+  binaries from scratch. These configurations are suitable as examples and for
+  one-off testing.
+
+- The template configurations rely on prebuilt worker images (built before
+  starting the tests). Prebuilt images include the gRPC binary, and are suitable
+  for running a batch of tests on the same gRPC version, or for running the same
+  test repeatedly.
+
+In addition to the controller, the [Test Infra] repository contains a set of
+[tools], including a test runner and tools to build and delete prebuilt worker
+images, as well as a [dashboard] implementation.
```

Comment:
The `[dashboard]` link isn't defined.

Btw, as a general rule, follow such markdown-link syntax with `[]` like this: `[dashboard][]`. This allows IDEs like VS Code (including the cloud versions) to give you edit-time feedback whether the link definition is missing or not. I also makes it easier to grep for missing link definitions while reducing false positives.

Reply:
Done.